src/chips/merkle_sum_tree.rs

```diff
@@ -287,7 +293,31 @@ impl MerkleSumTreeChip {
         // 3. Constrain the digest to be equal to the hash of the left and right values
         let computed_hash = poseidon_chip.hash(
             layouter.namespace(|| "hash four child nodes"),
-            [left_hash, left_balance, right_hash, right_balance],
+            [
+                left_hash,
+                left_balance.clone(),
+                right_hash,
+                right_balance.clone(),
+            ],
+        )?;
+
+        // Initiate the overflow check chip
+        let overflow_chip = OverflowChip::construct(self.config.overflow_check_config.clone());
+
+        overflow_chip.load(&mut layouter)?;
+
+        // Each balance cell is constrained to be less than the maximum limit
+        overflow_chip.assign(
```

Comment:
I think that there's an issue here. There's no constraint in place that forces `update_value` to be equal to `left_balance` or to `right_balance`. I think this can be resolved by modifing assign function inside the overflow_chip to take `update_value: &AssignedCell<Fp, Fp>` as input and copying the assigned cell to `self.config.a`. This is a similar operation to the one performed in the `enforce_less_than()` function.

Reply:
Ah, indeed! I completely overlooked this section! Thanks for bringing it to my attention
 
Fixed at [4a42627](https://github.com/summa-dev/summa-solvency/pull/33/commits/4a42627ac6bb8b1d628dc5ca9ac2deedc4d862ac)